src/web_resources/HangarXPLOR.Billing.js

```diff
@@ -0,0 +1,88 @@
+var HangarXPLOR = window.HangarXPLOR || {};
+HangarXPLOR.Billing =  window.HangarXPLOR.Billing || {};
+
+HangarXPLOR.Billing.Bills = [];
+HangarXPLOR.Billing.DoneLoading = false;
+
+HangarXPLOR.Billing.LoadData = function() {
+    HangarXPLOR.Billing.parseBills(1);
+};
+
+HangarXPLOR.Billing.getBill = function(name, element) {
+    let date = element.getElementsByClassName('date-col')[0].lastChild.data.trim();
+    const bills = HangarXPLOR.Billing.Bills[HangarXPLOR.Billing.parseDate(date)];
+
+    if(typeof bills === 'undefined') {
+        return null;
+    }
+
+    if(bills.length == 1) {
+        return bills[0];
+    } else if(bills.length > 1) {
+        for(let i = 0; i < bills.length; i++) {
+            if(bills[i]['name'].includes(name.toLowerCase())) {
+                return bills[i];
+            }
+        }
+
+        let fuse = new Fuse(bills, {keys: ['name']})
+        let results = fuse.search(name);
+
+        if(results.length > 0) {
+            return results[0];
+        }
+    }
```

Comment:
How big is the hangar you're testing with?

I know some people have a few thousand items in their hangar, and grey-marketeers also have considerable invoice history - just some things to consider. I haven't had a chance to do a full review yet, just doing a quick read over the code before bed :)

Reply:
~50ish items. - I agree that this is not ideal but I don't think there are any other hints then the name and the date to match an item with an invoice, and considering that this gets only called once when they open a specific item this should not be too big of a problem. But it would be great if someone with a large hangar could test the performance.